### ISSUE TITLE
Boolean checkboxes are broken

### ISSUE THREAD

**Issue by darioteixeira** [ocsigen/eliom#185]
Boolean checkboxes are currently broken.  The problem seems to be that the browser sends `on` when a checkbox is checked, but Eliom uses `bool_of_string` to decipher the boolean value, which of course expects `true` for true values. Below is a minimal example which replicates the problem:

````ocaml
open Eliom_content.Html5.F

let coucou_handler b () =
    Lwt.return
        (html
            (head (title (pcdata "Coucou")) [])
            (body [p [pcdata "You sent: "; pcdata (string_of_bool b)]]))

let coucou_service =
    Eliom_registration.Html5.register_service
        ~path:["coucou"]
        ~get_params:Eliom_parameter.(bool "b")
        coucou_handler

let coucou_form e_b =
    [
    label ~a:[a_for e_b] [pcdata "Is it true?"];
    bool_checkbox ~name:e_b ();
    button ~button_type:`Submit [pcdata "Submit"];
    ]

let main_handler () () =
    Lwt.return
        (html
            (head (title (pcdata "Main")) [])
            (body
                [
                h1 [pcdata "Links for coucou:"];
                p [a coucou_service [pcdata "True"] true];
                p [a coucou_service [pcdata "False"] false];
                h1 [pcdata "Forms for coucou:"];
                get_form coucou_service coucou_form;
                ]))

let main_service =
    Eliom_registration.Html5.register_service
        ~path:[]
        ~get_params:Eliom_parameter.unit
        main_handler
````

**Comment by vasilisp** [comment]
For what is worth, the form works with the sharedreact branch, but the links are broken. In sharedreact, I believe the [mere existence of the parameter](https://github.com/ocsigen/eliom/blame/sharedreact/src/lib/eliom_parameter.server.ml#L269) (even if it is the string "false") is interpreted as "true".

**Comment by vasilisp** [comment]
The fix made it to master some time ago (962fe5547f13fae61e4c18c718186728fbea600a), so closing.
